craft_application/services/fetch.py

```diff
@@ -104,21 +106,22 @@ def set_policy(self, policy: typing.Literal["strict", "permissive"]) -> None:
         """Set the policy for the fetch service."""
         self._session_policy = policy
 
-    def create_session(self, instance: craft_providers.Executor) -> dict[str, str]:
-        """Create a new session.
-
-        :return: The environment variables that must be used by any process
-          that will use the new session.
-        """
+    def create_session(self, instance: craft_providers.Executor) -> None:
```

Comment:
This one is more of an API change. We could either have it return that dict anyway or call it a major release. (In practice I don't think anything's using this)

Reply:
Same response as https://github.com/canonical/craft-application/pull/823#discussion_r2229599784

Reply:
It's returning a dict again due to https://github.com/canonical/craft-application/pull/823/commits/8d89177c127def2238600f1deb25ca1e49312c1e